### PR TITLE
Add Cstruct.rev to build a reversed cstruct

### PR DESCRIPTION
**Pull request by emillon**
It's like `List.rev`, but for cstructs. Let me know what you think.

### CODE REVIEW

File: lib_test/tests.ml [hunk 1]
```diff
@@ -158,6 +170,11 @@ let suite = [
     "aligned to 512"  , `Quick, check_alignment 512;
     "aligned to 0"    , `Quick, check_alignment_zero;
     "aligned to large", `Quick, check_alignment_large;
+  ];
+  "rev", [
+    "empty", `Quick, rev_empty;
+    "len = 1", `Quick, rev_len_1;
+    "len = 5", `Quick, rev_len_5;
```
Comment (avsm) [inline lib_test/tests.ml]: any chance of a len=0 test for completeness?

**Comment by avsm** [comment]
looks like it does what it says on the tin! :) happy to merge this

File: lib/cstruct.mli [hunk 2]
```diff
@@ -501,3 +501,6 @@ val append: t -> t -> t
 val concat: t list -> t
 (** [concat ts] is the concatenation of all the [ts]. It is not guaranteed that
  * the result is a newly created [t] in the zero- and one-element cases. *)
+
+val rev: t -> t
+(** [rev t] is [t] in reverse order. *)
```
Comment (avsm) [inline lib/cstruct.mli]: perhaps clarify that this *allocates* a fresh [t] and does not mutate the input

File: lib_test/tests.ml (hunk 1 above)
Comment (emillon) [reply lib_test/tests.ml]: That's what the `rev_empty` test does. Do you mean it should be renamed?

File: lib/cstruct.mli (hunk 2 above)
Comment (emillon) [reply lib/cstruct.mli]: Good idea! Done.

File: lib_test/tests.ml (hunk 1 above)
Comment (avsm) [reply lib_test/tests.ml]: missed that, thanks!

**Pull request merged**